### PR TITLE
Luminosity in data definitions

### PR DESCRIPTION
**Pull request by StivlWrith**
Add luminosityclass in DataDefinitions and tools for it to write and readwhen needed

### CODE REVIEW

File: DataDefinitions/StarClass.cs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿using MathNet.Numerics.Distributions;
+using MathNet.Numerics.Distributions;
```
Comment (richardbuckle) [inline DataDefinitions/StarClass.cs]: Why is this added?

File: DataDefinitions/Body.cs [hunk 2]
```diff
@@ -51,7 +51,7 @@ public class Body
         public string stellarclass;
 
         /// <summary>The Luminosity Class of the Star (since 2.4)</summary>
-        public string luminosityClass { get; set; }
+        public string luminosityclass { get; set; }
```
Comment (richardbuckle) [inline DataDefinitions/Body.cs]: Need to check that changing the case here doesn't break anything

File: EDDI/EDDI.cs [hunk 3]
```diff
@@ -1234,6 +1234,1664 @@ private bool eventStarScanned(StarScannedEvent theEvent)
                 star.stellarclass = theEvent.stellarclass;
                 star.solarmass = theEvent.solarmass;
                 star.solarradius = theEvent.solarradius;
+                star.luminosityclass = theEvent.luminosityclass;
```
Comment (richardbuckle) [inline EDDI/EDDI.cs]: This is a much bigger change than I expected

File: Events/StarScannedEvent.cs [hunk 4]
```diff
@@ -123,3 +123,79 @@ public StarScannedEvent(DateTime timestamp, string name, string stellarclass, de
         }
     }
 }
+
```
Comment (richardbuckle) [inline Events/StarScannedEvent.cs]: Again much more change than I expected just to add luminosity class

**Comment by richardbuckle** [comment]
On hold as per the originatig issue #133

File: DataDefinitions/StarClass.cs (hunk 1 above)
Comment (StivlWrith) [reply DataDefinitions/StarClass.cs]: I don't touch at that

File: DataDefinitions/Body.cs (hunk 2 above)
Comment (StivlWrith) [reply DataDefinitions/Body.cs]: all others are in lowercase, so i think that only one with uppercase inside was not very good

File: EDDI/EDDI.cs (hunk 3 above)
Comment (StivlWrith) [reply EDDI/EDDI.cs]: I don't undestand what you want to say

File: DataDefinitions/StarClass.cs (hunk 1 above)
Comment (richardbuckle) [reply DataDefinitions/StarClass.cs]: You must have touched it inadvertently.

File: DataDefinitions/Body.cs (hunk 2 above)
Comment (richardbuckle) [reply DataDefinitions/Body.cs]: You can't just go renaming things without checking what refers to that name.

File: EDDI/EDDI.cs (hunk 3 above)
Comment (richardbuckle) [reply EDDI/EDDI.cs]: You have added 1660 lines of code. ONE THOUSAND SIX HUNDRED AND SIXTY lines of code. Did you not notice that before you submitted your PR?

**Comment by Tkael** [comment]
The event already includes luminosityclass. The source data for BodyDetails doesn't include this data so these changes don't seem like they will help? or is the intent simply to make sure this data is passed to EDDN?

**Comment by StivlWrith** [comment]
Actually the data is collected in the event, we can read it in the event but it is not in ReportBody, or in system.bodies.
To have, it is needed to create the place inside the body and to send the data from the event to the body.

Have only two or one of the three is some useless except if you accept to use only script to speech the luminosity while using 
in Star scanned
{SetState('lum_class', Spacialise(event.luminosityclass))}
and in Star report
{state.lum_class}

**Comment by richardbuckle** [comment]
Superseded by PR #151
